content/features/extensibility/event-system/index.md

```diff
@@ -9,23 +9,102 @@ dnnversion:
 
-## Theme (Skin) Events
+The areas within DNN that you can hook into are:
 
-Theme events happen during the server side skin load lifecycle of any page when it is being accessed.
+- Theme (Skin) Events
+- Container Events 
+- Page (Tab) Events
+- Module Events
+- User Events
+- User Relationship Events
+- File Events
+- Folder Events
+- Role Events
+- Portal Events
 
-You can attach code to the following Theme events: 
+## Skin (Theme) Events
+
+Skin Events happen during the server-side skin load lifecycle of any page when it is being accessed.
+
+You can attach Skin Event Listeners to the following Skin Events: 
+
+- OnSkinInit
+- OnSkinLoad
+- OnSkinPreload
+- OnSkinUnload
+
+For each type of Skin Event Listener, there are a minimum of two parts:
+ 1. Registering the Skin Event Listener
+ 2. Implementing the Skin Event Listener
+
+The following example shows how to create an Event Listener for the OnSkinLoad event. 
+
+Note: The steps to create the other Skin Event Listeners are the same except your will change the SkinEventType in the sample code below.
+
+To attach code to the Skin Event, you must first register your Skin Event Listener. In this example we will register our Skin Event Listener when DNN starts loading for the first time. 
+
+There are several ways to accomplish this, the following example registers the events within a Web API route mapper since that code loads when DNN starts up.
+
+    using DotNetNuke.Application;
+    using DotNetNuke.UI.Skins.EventListeners;
+    using DotNetNuke.Web.Api;
+    
+    namespace SampleProject.DnnEvents.SkinEvents
+    {
+        class RouteMapper : IServiceRouteMapper
+        {
+    
+            public void RegisterRoutes(IMapRoute mapRouteManager)
+            {
+    
+                // Add Your Web API Routes Here
+    
+                // Register Your Skin Events
+    
+                DotNetNukeContext.Current.SkinEventListeners.Add(new DotNetNuke.UI.Skins.EventListeners.SkinEventListener(SkinEventType.OnSkinLoad, SampleSkinEventListener.OnSkinLoad));
+    
+            }
+    
+        }
+    }
+
+
+In the above example, we are registering a Skin Event Listener called **SampleSkinEventListener.OnSkinLoad**. This is a static class and method that defines what code will run when the Skin OnLoad event triggers. 
+
+Since all pages in a DNN website have a skin/theme, we can assume that this code will run on page load for every page within a DNN website.
+
+The following code is the Skin Event Listener implementation.   
+
+
+    using System;
+    using DotNetNuke.Services.Exceptions;
+    using DotNetNuke.UI.Skins.EventListeners;
+    
+    namespace SampleProject.DnnEvents.SkinEvents
+    {
+        public static class SampleSkinEventListener
+        {
+            public static void OnSkinLoad(object sender, SkinEventArgs e)
```

Comment:
Are there any interesting properties of `SkinEventArgs` that it's worth calling out?

Reply:
If it's Ok with you, I'll make a note on my end to look to see if I can provide some additional detail on my second pass through the page as I build it out a bit more. For now, I think they are self-explanatory for the audience that will be using them at this point.  As well, I'll try to come up with a supporting blog post about how/why a dev might use these? I can pull together my info from Southern Fried that I did the other week on this topic which had real world examples.